lib/auth/crownjewel/crownjewelv1/service_test.go

```diff
@@ -0,0 +1,221 @@
+			// test the method with allowed admin states, each one separately.
+			t.Run("allowed admin states", func(t *testing.T) {
+				for _, state := range tt.allowedStates {
+					t.Run(stateToString(state), func(t *testing.T) {
+						for _, verbs := range utils.Combinations(tt.allowedVerbs) {
+							t.Run(fmt.Sprintf("verbs=%v", verbs), func(t *testing.T) {
+								service := newService(t, state, fakeChecker{allowedVerbs: verbs})
+								err := callMethod(t, service, tt.name)
+								// expect access denied except with full set of verbs.
+								if len(verbs) == len(tt.allowedVerbs) {
+									require.False(t, trace.IsAccessDenied(err))
+								} else {
+									require.True(t, trace.IsAccessDenied(err), "expected access denied for verbs %v, got err=%v", verbs, err)
+								}
+							})
+						}
+					})
+				}
+			})
+
+			// test the method with disallowed admin states; expect failures.
+			t.Run("disallowed admin states", func(t *testing.T) {
+				disallowedStates := otherAdminStates(tt.allowedStates)
+				for _, state := range disallowedStates {
+					t.Run(stateToString(state), func(t *testing.T) {
+						// it is enough to test against tt.allowedVerbs,
+						// this is the only different data point compared to the test cases above.
+						service := newService(t, state, fakeChecker{allowedVerbs: tt.allowedVerbs})
+						err := callMethod(t, service, tt.name)
+						require.True(t, trace.IsAccessDenied(err))
+					})
+				}
+			})
+		})
+	}
+
+	// verify that all declared methods have matching test cases
+	t.Run("verify coverage", func(t *testing.T) {
+		for _, method := range crownjewelv1.CrownJewelService_ServiceDesc.Methods {
+			t.Run(method.MethodName, func(t *testing.T) {
+				match := false
+				for _, testCase := range testCases {
+					match = match || testCase.name == method.MethodName
+				}
+				require.True(t, match, "method %v without coverage, no matching tests", method.MethodName)
+			})
+		}
+	})
+}
+
+var allAdminStates = map[authz.AdminActionAuthState]string{
+	authz.AdminActionAuthUnauthorized:         "Unauthorized",
+	authz.AdminActionAuthNotRequired:          "NotRequired",
+	authz.AdminActionAuthMFAVerified:          "MFAVerified",
+	authz.AdminActionAuthMFAVerifiedWithReuse: "MFAVerifiedWithReuse",
+}
+
+func stateToString(state authz.AdminActionAuthState) string {
+	str, ok := allAdminStates[state]
+	if !ok {
+		return fmt.Sprintf("unknown(%v)", state)
+	}
+	return str
+}
+
+// otherAdminStates returns all admin states except for those passed in
+func otherAdminStates(states []authz.AdminActionAuthState) []authz.AdminActionAuthState {
+	var out []authz.AdminActionAuthState
+	for state := range allAdminStates {
+		found := slices.Index(states, state) != -1
+		if !found {
+			out = append(out, state)
+		}
+	}
+	return out
+}
+
+// callMethod calls a method with given name in the DatabaseObjectService service
+func callMethod(t *testing.T, service *Service, method string) error {
```

Comment:
```suggestion
func callMethod(t *testing.T, service crownjewelv1.CrownJewelServiceServer, method string) error {
```

Reply:
This breaks compilation and I don't want to update all types now, as we do that in all other services.